awx/main/db/profiled_pg/base.py

```diff
@@ -0,0 +1,101 @@
+import os
+import traceback
+import uuid
+
+from django.core.cache import cache
+from django.core.cache.backends.locmem import LocMemCache
+from django.db.backends.postgresql.base import DatabaseWrapper
+
+from awx.main.utils import memoize
+
+__loc__ = LocMemCache(str(uuid.uuid4()), {})
+__all__ = ['DatabaseWrapper']
+
+
+class RecordedQueryLog(object):
+
+    dest = '/var/lib/awx/profile'
+
+    def __init__(self, log, db):
+        self.log = log
+        self.db = db
+        self.threshold = cache.get('awx-profile-sql-threshold')
+
+    def append(self, query):
+        ret = self.log.append(query)
+        self.write(query)
+        return ret
+
+    def write(self, query):
+        if self.threshold is None:
+            return
+        seconds = float(query['time'])
+        if seconds >= self.threshold:
+            sql = query['sql']
+            if sql.startswith('EXPLAIN'):
+                return
+            if not os.path.isdir(self.dest):
+                os.makedirs(self.dest)
+            filename = '%.3fs-%s.sql' % (seconds, uuid.uuid4())
+            filepath = os.path.join(self.dest, filename)
+            bt = ' '.join(traceback.format_stack())
+            explain = ''
+            cursor = self.db.cursor()
+            cursor.execute('EXPLAIN ANALYZE VERBOSE {}'.format(sql))
```

Comment:
Maybe we _shouldn't_ `ANALYZE`, given that it actually executes the query again (and we know it to be slow).

Reply:
True that. Remove it!